client.go

```diff
@@ -168,70 +164,33 @@ func (c *Client) Dial(ctx context.Context) error {
 	if c.sechan != nil {
 		return errors.Errorf("secure channel already connected")
 	}
+
 	var err error
 	c.conn, err = uacp.Dial(ctx, c.endpointURL)
 	if err != nil {
 		return err
 	}
+
 	c.sechan, err = uasc.NewSecureChannel(c.endpointURL, c.conn, c.cfg)
 	if err != nil {
 		_ = c.conn.Close()
 		return err
 	}
 
-	// Issue #313: decouple the dial context from the monitor context
-	// mctx must *not* be a child context of 'ctx'. Otherwise, the
-	// monitor go routine terminates whenever the dial context is done
-	// which may get triggered unexpectedly by a timer context.
-	var mctx context.Context
-	mctx, c.cancelMonitor = context.WithCancel(context.Background())
-	go c.monitorChannel(mctx)
-	return c.openSecureChannel(mctx, c.sechan.Open)
-}
-
-func (c *Client) openSecureChannel(ctx context.Context, open func() error) error {
-	if err := open(); err != nil {
-		c.cancelMonitor()
-		_ = c.conn.Close()
-		c.sechan = nil
-		return err
-	}
-	c.scheduleRenewingToken(ctx)
-	return nil
-}
-
-func (c *Client) monitorChannel(ctx context.Context) {
-	for {
-		select {
-		case <-ctx.Done():
-			return
-		default:
-			msg := c.sechan.Receive(ctx)
-			if msg.Err != nil {
-				if msg.Err == io.EOF {
-					debug.Printf("Connection closed")
-				} else {
-					debug.Printf("Received error: %s", msg.Err)
-				}
-				// todo (dh): apart from the above message, we're ignoring this error because there is nothing watching it
-				// I'd prefer to have a way to return the error to the upper application.
-				return
-			}
-			debug.Printf("Received unsolicited message from server: %T", msg.V)
-		}
-	}
+	return c.sechan.Open(ctx)
 }
 
 // Close closes the session and the secure channel.
 func (c *Client) Close() error {
+	defer c.conn.Close()
+
 	// try to close the session but ignore any error
 	// so that we close the underlying channel and connection.
 	_ = c.CloseSession()
-	if c.cancelMonitor != nil {
-		c.cancelMonitor()
-	}
 
-	return c.sechan.Close()
+	 _ = c.sechan.Close()
```

Comment:
yeah, i'm torn as well. technically I should maybe wrap or append the first error. But no caller would ever check that.

Logging it is possible, but I think we should revisit logging and make it "first class" rather than the `debug.Log` method we have now.